### PR TITLE
Improve parsing of decorated new

### LINKED ISSUE #12886: `new` accepts values rather than types?
In Chapel 1.19.0, the following code seems to compile and run:

```chapel
class C {
  var x: int;
}

var myC = new owned C(42);
var myC2 = new owned myC;  // note: Calling `new` on a value, not a type.
writeln(myC2);
```

which suggests that `new owned` accepts not just type specifications, but also values.  The same is true for `new shared`.  This seems like code that the compiler shouldn't accept and I suspect it's just sneaking through somehow.  Interestingly, `new borrowed` and `new unmanaged` don't support this, and `new myC` generates a different error message about there being no arguments to the type (which is questionable, since it's not a type).

**Associated Future Test(s):**
[`test/classes/errors/newValueNotType.chpl`](
           https://github.com/chapel-lang/chapel/blob/master/test/classes/errors/newValueNotType.chpl
          ) #12946 

### Configuration Information

- Output of `chpl --version`: `chpl version 1.20.0 pre-release (f421723)`

### PR DESCRIPTION
**Pull request by mppf**
Resolves #12886

Related to PR #13932 and to issues #13957 #13950 and #14300.

This PR adjusts the parser to parse the following 3 cases differently:
``` chapel
new owned X(); // most common
new owned(X); // owned taking over an unmanaged ptr, say
new owned X; // always an error
```

Before this PR, the parser created the same AST representation for all 3, 
which made it challenging to make one of them into a compilation error.

- [x] rebuild parser
- [x] full local testing

Reviewed by @vasslitvinov - thanks!

### CODE REVIEW

File: compiler/parser/chapel.ypp [hunk 1]
```diff
@@ -220,6 +221,7 @@
   #include "stringutil.h"
   #include "TryStmt.h"
   #include "vec.h"
+  #include "wellknown.h"
```
Comment (vasslitvinov) [inline compiler/parser/chapel.ypp]: Please check if you can do without `wellknown.h`

File: modules/packages/LinearAlgebraJama.chpl [hunk 2]
```diff
@@ -2162,9 +2162,9 @@ class Matrix {
       return new unmanaged SingularValueDecomposition(_to_unmanaged(this));
    }
 
-   proc rsvd() {
+   /*proc rsvd() {
```
Comment (vasslitvinov) [inline modules/packages/LinearAlgebraJama.chpl]: Remove it altogether?

File: spec/Errors.tex [hunk 3]
```diff
@@ -546,6 +552,9 @@ \subsection{forall}
 if error(s) are thrown in the loop body.
 
 \begin{chapelexample}{handle-from-forall.chpl}
+\begin{chapelpre}
+class DemoError : Error { }
+\end{chapelpre}
```
Comment (vasslitvinov) [inline spec/Errors.tex]: Did you update #14165 with the above?

File: test/classes/errors/newValueNotType2.good [hunk 4]
```diff
@@ -1,6 +1,2 @@
-newValueNotType2.chpl:7: error: cannot use decorator borrowed on a value
-newValueNotType2.chpl:7: error: duplicate decorators - borrowed owned C
-newValueNotType2.chpl:7: error: invalid use of 'new'
-newValueNotType2.chpl:10: error: cannot use decorator unmanaged on a value
-newValueNotType2.chpl:10: error: duplicate decorators - unmanaged owned C
-newValueNotType2.chpl:10: error: invalid use of 'new'
+newValueNotType2.chpl:7: error: type in 'new' expression is missing its argument list
+newValueNotType2.chpl:10: error: type in 'new' expression is missing its argument list
```
Comment (vasslitvinov) [inline test/classes/errors/newValueNotType2.good]: Ugh this is unfortunate. I am OK with it nonetheless.

File: test/parsing/precedence/bang-vs-new-etc.chpl [hunk 5]
```diff
@@ -68,7 +68,7 @@ proc main {
   var z05 = new owned (tm())();
   check("z05 = new owned (tm())()", z05, "owned C");
 
-  var z06 = new (/*owned*/ tm())()?; // workaround for #13950
+  var z06 = new (owned tm())()?; // workaround for #13950
```
Comment (vasslitvinov) [inline test/parsing/precedence/bang-vs-new-etc.chpl]: If this is no longer a workaround, please remove the comment.

File: test/classes/errors/newValueNotType2.good (hunk 4 above)
Comment (mppf) [reply test/classes/errors/newValueNotType2.good]: I think this is the right error for this case. If the parens are added, it gives an error about not having a type, e.g.

```
newValueNotType4.chpl:9: error: 'new' must be followed by a type expression
```

I will add a sibling newValueNotType4.chpl test with the parens to make sure we notice if this changes.

**Comment by bradcray** [comment]
As noted on #14300, I think we should ultimately make `new owned(X)` an error as well.

**Pull request merged**